lib/bolt/node/ssh.rb

```diff
@@ -33,6 +33,8 @@ def connect
                                     Net::SSH::Verifiers::Secure.new
                                   end
       options[:timeout] = @connect_timeout if @connect_timeout
+      # Only use ssh-agent if it is configured and its socket exists.
+      options[:use_agent] = ENV['SSH_AUTH_SOCK'].to_s && File.socket?(ENV['SSH_AUTH_SOCK'])
```

Comment:
lets add
```
@logger.debug("Could not connect to ssh agent.")
```

Reply:
Does checking the socket work on Windows? I’m not sure if putty’s ssh agent uses named pipes?

Reply:
I see now that we shouldn't do File.socket validation on the value of SSH_AUTH_SOCK. If the user specifies an invalid value for SSH_AUTH_SOCK, that should be net-ssh's responsibility.

net-ssh `use_agent` defaults to true, and when true, net-ssh agent `connect()` expects an agent to be configured. Since we don't allow configuration of `use_agent` within bolt, we could implement the following checks within bolt to avoid the "Agent not configured" or "pageant process not running" errors:

https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/authentication/agent.rb#L85
https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/authentication/pageant.rb#L403

Amended PR, with debugging. Thanks, all.